open_spiel/python/policy.py

```diff
@@ -268,7 +265,13 @@ def policy_for_key(self, key):
       tabular_policy.policy_for_key(s)[:] = [0.1, 0.5, 0.4]
       ```
     """
-    return self.action_probability_array[self.state_lookup[key]]
+    try:
+      return self.action_probability_array[self.state_lookup[key]]
+    except KeyError:
+      # In case they are no legal actions in a state, then the state
+      # will not be added to action_probability_array in the initialization,
+      # therefore 0 should be played.
+      return {0: 1}
```

Comment:
I will update this part of the code once https://github.com/deepmind/open_spiel/pull/625/files get merged.